course6/toyapp_alexs/displaylib/vector.h

```diff
@@ -0,0 +1,79 @@
+#pragma once
+
+#include <array>
+
+namespace Display
+{
+template<class T, std::size_t S>
+struct Vector
+{
+public:
+	template<class... E>
+	Vector(E... data)
+		: mData {static_cast<T>(data)...}
+	{
+	}
+
+	T& operator [](int index)
+	{
+		return mData[index];
+	}
+private:
+	std::array<T, S> mData;
+};
+
+struct Pixel: public Vector<int, 2>
+{
+	Pixel()
+		: Vector(),
+		  x((*this)[0]),
+		  y((*this)[1])
+	{}
+
+	Pixel(int x, int y)
+		: Vector(x, y),
+		  x((*this)[0]),
+		  y((*this)[1])
+	{}
+
+	Pixel& operator =(const Pixel& other)
+	{
+		this->x = other.x;
+		this->y = other.y;
+		return *this;
+	}
+
+	int& x;
```

Comment:
Would be easier if you'd eschew the public fields, and provide getters ```x(); y()``` instead. 
Even better if you just acknowledge that you an always have API that describes the world precisely, and just have ```operator[]```.